File: LeetCode/1944. Number of Visible People in a Queue/solution.py
class Solution:
    def canSeePersonsCount(self, heights: List[int]) -> List[int]:
        st = []
        n = len(heights)
        ans = [0]*n 
        for i in range(n)[::-1]:
            cnt = 0
            while st and heights[i] > st[-1]:
                cnt += 1
                st.pop()
            if st:
                cnt += 1
            ans[i] = cnt 
            st.append(heights[i])
            
        return ans

File: LeetCode/1944. Number of Visible People in a Queue/solution.cpp
class Solution
{
public:
    vector<int> canSeePersonsCount(vector<int> &heights)
    {
        stack<int> st;
        // cout << std::numeric_limits<int>::infinity() << endl;
        st.push(std::numeric_limits<int>::max());
        int n = heights.size();
        vector<int> ans(n);
        // cout << st.size() << endl;
        for (int i = n - 1; i >= 0; i--)
        {
            int cnt = 0;
            while (st.top() < heights[i])
            {
                cnt++;
                st.pop();
            }
            ans[i] = cnt+(st.top() != std::numeric_limits<int>::max());
            st.push(heights[i]);
            
        }
        return ans;
    }
};